scripts/verify_repository_hygiene.py

```
@@ -0,0 +1,86 @@
+FORBIDDEN_PARTS = {".venv", "venv", "node_modules", ".next", ".pytest_cache", "__pycache__", ".work", "backup", "backups"}
+FORBIDDEN_SUFFIXES = (".pyc", ".log", ".db", ".sqlite", ".sqlite3", ".tsbuildinfo", ".bak")
+CONFLICT = re.compile(r"^(<<<<<<<|=======|>>>>>>>)", re.MULTILINE)
+
+
+SOURCE_MANIFEST = ".wfh-source-manifest"
+
+
+def tracked(root: Path) -> list[str]:
+    git = shutil.which("git")
+    if git is not None and (root / ".git").exists():
+        out = subprocess.check_output([git, "ls-files"], cwd=root, text=True)
```

Ruff (0.16.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)